rpc/src/module/ipc.rs

```
@@ -0,0 +1,883 @@
+
+        // In any case, we will close all pipes after a certain period of time.
+        // This ensures that the function always completes if the ipc script is not written as expected.
+        let pipesfin = pipesctx.clone();
+        let mut signal_timeout = tokio::sync::watch::channel(0);
+        let limit_time = self.limit_time;
+        self.shared.async_handle().spawn({
+            async move {
+                let _ = tokio::time::timeout(
+                    tokio::time::Duration::from_secs(limit_time as u64),
+                    signal_timeout.1.changed(),
+                )
+                .await;
+                pipesfin.close();
+            }
+        });
+
+        let pipesfin = pipesctx.clone();
+        let limit_cycles = self.limit_cycles;
+        let upper_cycles = 1000000;
+        self.shared.async_handle().spawn({
+            async move {
+                // Ignore its return value as it is unimportant.
+                let _ = script_verifier
+                    .create_scheduler(&script_group)
+                    .map(|mut scheduler| {
+                        // Make the vm can always be stopped.
+                        debug_assert_eq!(scheduler.consumed_cycles(), 0);
+                        let tic = std::time::SystemTime::now();
+                        loop {
+                            let last_cycles = limit_cycles as u64 - scheduler.consumed_cycles();
+                            let step_cycles = std::cmp::min(upper_cycles, last_cycles);
+                            let result = scheduler.run(RunMode::LimitCycles(step_cycles as u64));
+                            if let Err(ckb_vm::Error::CyclesExceeded) = result {
+                                if step_cycles < upper_cycles {
+                                    break;
+                                }
+                                let toc = tic.elapsed().map(|e| e.as_secs()).unwrap_or(u64::MAX);
+                                if toc >= limit_time as u64 {
```

if timeout, whether we should report an error for endpoint?